x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -157,7 +160,15 @@ public void execute(EsqlQueryRequest request, EsqlExecutionInfo executionInfo, P
             new EsqlSessionCCSUtils.CssPartialErrorsActionListener(executionInfo, listener) {
                 @Override
                 public void onResponse(LogicalPlan analyzedPlan) {
-                    executeOptimizedPlan(request, executionInfo, planRunner, optimizedPlan(analyzedPlan), listener);
+                    try {
+                        queryBuilderResolver.resolveQueryBuilders(
+                            optimizedPlan(analyzedPlan),
```

Comment:
while here, and my comment predates ur changes, would u mind to pull `optimizedPlan(analyzedPlan)` out into a separate variable, so that it is more straightforward to read, e.g.

```
var optimizedPlan = optimizedPlan(analyzedPlan)
```